internal/controller/istiogatewaysecret/controller_test.go

```diff
@@ -0,0 +1,124 @@
+package istiogatewaysecret_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	apicorev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/kyma-project/lifecycle-manager/internal/controller/istiogatewaysecret"
+)
+
+func TestReconcile_WhenGetSecretFuncReturnsError_ReturnError(t *testing.T) {
+	// ARRANGE
+	var stubGetterFunc istiogatewaysecret.GetterFunc = func(ctx context.Context, name types.NamespacedName) (*apicorev1.Secret, error) {
+		return nil, errors.New("some-error")
+	}
+	mockHandler := &mockHandler{}
+	reconciler := istiogatewaysecret.NewReconciler(stubGetterFunc, mockHandler)
+
+	// ACT
+	_, err := reconciler.Reconcile(context.TODO(), ctrl.Request{})
+
+	// ASSERT
+	require.ErrorContains(t, err, "failed to get istio gateway root secret: some-error")
+	assert.Equal(t, 0, mockHandler.calls)
+}
+
+func TestReconcile_WhenGetSecretFuncReturnsNoErrorAndSecretIsNil_ReturnError(t *testing.T) {
+	// ARRANGE
+	var stubGetterFunc istiogatewaysecret.GetterFunc = func(ctx context.Context, name types.NamespacedName) (*apicorev1.Secret, error) {
+		return nil, nil
+	}
+	mockHandler := &mockHandler{}
+	reconciler := istiogatewaysecret.NewReconciler(stubGetterFunc, mockHandler)
+
+	// ACT
+	_, err := reconciler.Reconcile(context.TODO(), ctrl.Request{})
+
+	// ASSERT
+	require.ErrorIs(t, err, istiogatewaysecret.ErrSecretNotFound)
+	assert.Equal(t, 0, mockHandler.calls)
+}
+
+func TestReconcile_WhenGetSecretFuncIsCalled_IsCalledWithRequestNamespacedName(t *testing.T) {
+	// ARRANGE
+	secretName, secretNamespace := "test-name", "test-namespace"
+	var stubGetterFunc istiogatewaysecret.GetterFunc = func(ctx context.Context, name types.NamespacedName) (*apicorev1.Secret, error) {
+		assert.Equal(t, secretNamespace, name.Namespace)
+		assert.Equal(t, secretName, name.Name)
+		return &apicorev1.Secret{}, nil
+	}
+	reconciler := istiogatewaysecret.NewReconciler(stubGetterFunc, &mockHandler{})
+	request := ctrl.Request{NamespacedName: types.NamespacedName{Name: secretName, Namespace: secretNamespace}}
+
+	// ACT
+	// ASSERT
+	_, _ = reconciler.Reconcile(context.TODO(), request)
+}
+
+func TestReconcile_WhenGetSecretFuncReturnsSecret_HandlerManageGatewaySecretIsCalled(t *testing.T) {
+	// ARRANGE
+	secret := &apicorev1.Secret{}
```

Comment:
The previous one is asserting that the GetterFunc is called with the values of the request.
This one is asserting that the Handler impl is called and one time only.

I try to convey the different testcases by descriptive test names.
Let me know if they should be changed to something better!

Reply:
yes but I mean it is the same setup so it could be just one test for both? The test names convey the purpose so I am fine with keeping them separate. 
But in this case do we actually need the following lines in the previous test?

	request := ctrl.Request{NamespacedName: types.NamespacedName{Name: secretName, Namespace: secretNamespace}}

	// ACT
	// ASSERT
	_, _ = reconciler.Reconcile(context.TODO(), request)

Reply:
Hm yes you are right:I changed it to just return nil instead of a Secret and moved the request more in focus. I think it's better now 👍